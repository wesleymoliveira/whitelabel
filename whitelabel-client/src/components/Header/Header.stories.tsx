import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { HeaderProps } from '@/typings/shared';
import { Header } from './Header';

export default {
	title: 'components/Header',
	component: Header
} as Meta;

export const Default: Story<HeaderProps> = () => <Header />;

export const NoLeft: Story<HeaderProps> = () => <Header noLeftContent />;

export const NoRight: Story<HeaderProps> = () => <Header noRightContent />;

export const NoCenter: Story<HeaderProps> = () => <Header noCenterContent />;

export const HomeOnLeft: Story<HeaderProps> = () => <Header homeOnLeft />;

export const WithActions: Story<HeaderProps> = () => (
	<Header leftAction={() => alert('Left Action')} rightAction={() => alert('Left Action')} />
);
