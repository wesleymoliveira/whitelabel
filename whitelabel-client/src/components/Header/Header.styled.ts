import { Box, styled } from '@mui/material';
import { css } from '@emotion/react';

export const StyledBox = styled(Box)`
	${({ theme }) => css`
		display: flex;
		width: 100%;
		align-items: center;
		justify-content: space-between;
		border-bottom: 1px solid ${theme.palette.common.black};
		position: relative;
		min-height: 60px;
	`}
`;
