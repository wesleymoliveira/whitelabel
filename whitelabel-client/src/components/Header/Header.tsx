import Image from 'next/image';
import { Typography, Box } from '@mui/material';
import { HeaderProps } from '@/typings/shared';
import * as S from './Header.styled';
import { useRouter } from 'next/router';

export function Header({
	leftAction = () => undefined,
	rightAction = () => undefined,
	noCenterContent = false,
	noLeftContent = false,
	noRightContent = false,
	homeOnLeft = false
}: HeaderProps) {
	const { push } = useRouter();

	const handlePush = () => push('/home');

	return (
		<S.StyledBox p={homeOnLeft ? 2.5 : 1} data-testid='header' component='header'>
			{noLeftContent ? (
				<Box />
			) : homeOnLeft ? (
				<Image
					src='https://res.cloudinary.com/dgop0ic8b/image/upload/v1650298104/icons/home_asj9qz.svg'
					alt='Arrow Back'
					width='25'
					height='25'
					onClick={leftAction}
				/>
			) : (
				<Box pl={2} py={1} onClick={leftAction}>
					<Image
						src='https://res.cloudinary.com/dgop0ic8b/image/upload/v1650298097/icons/arrow-back_yz4824.svg'
						alt='Arrow Back'
						width='18'
						height='13'
						style={{ cursor: 'pointer' }}
					/>
				</Box>
			)}
			{!noCenterContent ? (
				<Box
					sx={{
						display: 'flex',
						flexDirection: 'column',
						alignItems: 'center',
						justifyContent: 'center',
						position: 'absolute',
						cursor: 'pointer',
						left: '50%',
						top: '50%',
						transform: 'translate(-50%, -50%)'
					}}
					onClick={handlePush}
				>
					<Typography lineHeight={1.1} fontSize={7} fontWeight={400} textTransform='uppercase'>
						A Charity Gala & Dinner for
					</Typography>
					<Typography lineHeight={1.1} fontSize={30} fontWeight={400} textTransform='uppercase'>
						UKRAINE
					</Typography>
					<Typography lineHeight={1.1} fontSize={8} fontWeight={400} textTransform='uppercase'>
						Benefitting the Red Cross
					</Typography>
				</Box>
			) : (
				<Box />
			)}
			{noRightContent ? (
				<Box />
			) : (
				<Box pr={2} py={1} onClick={rightAction}>
					<Image
						src='https://res.cloudinary.com/dgop0ic8b/image/upload/v1650298104/icons/heart1_spo1vl.svg'
						alt='Heart Menu'
						width='25'
						height='25'
						style={{ cursor: 'pointer' }}
					/>
				</Box>
			)}
		</S.StyledBox>
	);
}
