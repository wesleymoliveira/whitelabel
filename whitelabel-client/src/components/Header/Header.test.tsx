import { render, screen } from '@testing-library/react';

import { Header } from './Header';

describe('<Header />', () => {
	it('should render the component', () => {
		const { container } = render(<Header />);

		expect(screen.getByTestId(/header/i)).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});
});
