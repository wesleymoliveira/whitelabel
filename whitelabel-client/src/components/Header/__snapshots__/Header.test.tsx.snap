// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the component 1`] = `
.emotion-0 {
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid #000;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  line-height: 1.1;
  font-size: 7px;
  font-weight: 400;
  text-transform: uppercase;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  line-height: 1.1;
  font-size: 30px;
  font-weight: 400;
  text-transform: uppercase;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  line-height: 1.1;
  font-size: 8px;
  font-weight: 400;
  text-transform: uppercase;
}

<header
  class="MuiBox-root emotion-0"
  data-testid="header"
>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2718%27%20height=%2713%27/%3e"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      alt="Arrow Back"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <div
    class="MuiBox-root emotion-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-2"
    >
      A Charity Gala & Dinner for
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-3"
    >
      UKRAINE
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-4"
    >
      Benefitting the Red Cross
    </p>
  </div>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2725%27%20height=%2725%27/%3e"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      alt="Heart Menu"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
</header>
`;
